docs/data-cookbooks/run-tasks-in-lambda-or-docker.md

```diff
@@ -0,0 +1,99 @@
+
+## When to use Lambda
+
+You should use AWS Lambda whenever all of the following are true:
+
+* The task runs on one of the supported [Lambda Runtimes](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html). At time of this writing, supported runtimes include versions of python, Java, Ruby, node.js, Go and .NET.
+* The lambda package is less than 50 MB in size, zipped.
+* The task consumes less than each of the following resources:
+    * 3008 MB memory allocation
+    * 512 MB disk storage (must be written to `/tmp`)
+    * 15 minutes of execution time
+
+See [this page](https://docs.aws.amazon.com/lambda/latest/dg/limits.html) for a complete and up-to-date list of AWS Lambda limits.
+
+If your task requires more than any of these resources or an unsupported runtime, creating a Docker image which can be run on ECS is the way to go. Cumulus supports running any lambda package as a Docker container with [`cumulus-ecs-task`](https://github.com/nasa/cumulus-ecs-task).
+
+## Step Function Activities and `cumulus-ecs-task`
+
+[Step Function Activities](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-activities.html) enable a state machine task to "publish" an activity task which can be picked up by any activity worker. Activity workers can run pretty much anywhere, but Cumulus workflows support the [`cumulus-ecs-task`](https://github.com/nasa/cumulus-ecs-task) activity worker. The `cumulus-ecs-task` worker runs as a Docker container on the Cumulus ECS cluster.
+
+The `cumulus-ecs-task` container takes an AWS Lambda Amazon Resource Name (ARN) as an argument (see `--lambdaArn` in the example below). This ARN argument is defined at deployment time. The `cumulus-ecs-task` worker polls for new Step Function Activity Tasks. When a Step Function executes, the worker (container) picks up the activity task and runs the code contained in the lambda package defined on deployment.
+
+## Example: Replacing AWS Lambda with a Docker container run on ECS
+
+Whether using AWS Lambda or Step Functions Activities, there will be a lambda package defined in `lambdas.yml`:
+
+```yaml
+QueueGranules:
+  handler: index.handler
+  timeout: 900
+  memory: 3008
+  source: node_modules/@cumulus/queue-granules/dist/
+  useMessageAdapter: true
+```
+
+Given the `QueueGranules` lambda package is already used Step Function State Machine definition, the following will exist in a workflows yml file:
+
+```yaml
+    QueueGranules:
+      Type: Task
+      Resource: ${QueueGranulesLambdaFunction.Arn}
+      Next: StopStatus
+```
+
+Given it has been discovered this task can no longer run in AWS Lambda, it can be run on the Cumulus ECS cluster by including the following in `app/config.yml`:
+
+```yaml
+  ecs:
+    instanceType: t2.large
+    desiredInstances: 1
+    services:
+      QueueGranules:
+        docker: true
+        image: cumuluss/cumulus-ecs-task:1.2.5
```

Comment:
I assume this is the case, but is `cumuluss/` the right owner name for our Docker images? I've seen this and wondered if it was a typo. But I guess it's not since deployments are using this value right now

Reply:
Ya it's strange. Even stranger is it might have been a typo at the time, I vaguely remember @scisco telling me this. Or it could be that `cumulus` was taken. But yes `cumuluss` with 2 `s`'s is the Docker owner we are using at this time.